eng/pipelines/common/xplat-setup.yml

```diff
@@ -47,11 +48,11 @@ jobs:
 
       - name: _BuildConfig
         value: $(buildConfigUpper)
-      
+
       - name: _runSmokeTestsOnlyArg
-        ${{ if ne(parameters.variables['isFullMatrix'], true) }}:
+        ${{ if ne(variables['nonPRBuild'], true) }}:
           value: /p:RunSmokeTestsOnly=true
-        ${{ if eq(parameters.variables['isFullMatrix'], true) }}:
+        ${{ if eq(variables['nonPRBuild'], true) }}:
```

Comment:
You could've just used the isFullMatrix variable, no need to introduce a new one. That variable is defined globally as it is imported on the common variables template. Can we please clean that up in a separate PR?

Reply:
It didn't seem to work, that is why I had to add a new variable. But looking at the builds from today, even that didn't work. The latest build is still passing `/p:RunSmokeTestsOnly=true`: [Build](https://dev.azure.com/dnceng/public/_build/results?buildId=1447748&view=logs&j=108d2c4a-8a62-5a58-8dad-8e1042acc93c&t=568f884b-cc12-5fd3-e7fe-790b5ac403f4)

https://dev.azure.com/dnceng/public/_build?definitionId=686&_a=summary&repositoryFilter=310&branchFilter=149480%2C149480

What would be the correct fix?

Reply:
I think the check for `eq` check for true needs to be True.

If you look at the expanded yml from a build today, you'll see it. Well, you'll at least see `isFullMatrix: True`

https://dev.azure.com/dnceng/public/_apis/build/builds/1447748/logs?logid=1

Reply:
how did you get this? I wasn't able to find it.

> I think the check for eq check for true needs to be True.

All the instances in `eng/pipelines` are using lowercase `true`.

Reply:
Alex gave me the tip and I have it bookmarked.  

I'm just guessing based upon what I've seen.  It's a bit maddening if "True" ;-)

Reply:
The way you have it won't work because the variables you are defining haven't been defined. Those are defined later in the evaluation, as we pass them as parameters and then define it at the last node of the expansion tree (the deepest template). These conditions are compile time conditions and so the variable that you just defined will evaluate to null.

However isFullMatrix should be resolved here as it is defined in a variables template and those are expanded first and hence evaluate to the right value. That's why we are able to use it all over the place, I.e: look at helix-queues-setup.yml.

Is the behavior you are seeing not working for runtime.yml ?